### PR TITLE
[stable/grafana] Add support for multiple replicas and anti affinity to grafana

### PR DESCRIPTION
**Pull request by simontesar**
*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

This PR adds support for multiple replicas and anti affinity to the stable/grafana chart.

### CODE REVIEW

File: stable/grafana/values.yaml [hunk 1]
```diff
@@ -12,6 +12,14 @@ server:
   nodeSelector: {}
   tolerations: []
 
+  # Multiple replicas do not work with the default sqlite database,
+  # see http://docs.grafana.org/tutorials/ha_setup/.
+  replicaCount: 1
+
+  antiAffinity:
+    enabled: true
```
Comment (viglesiasce) [inline stable/grafana/values.yaml]: Can you disable this by default?

**Review by viglesiasce — CHANGES_REQUESTED** [review]

**Comment by simontesar** [comment]
Changed anti-affinity to false, @viglesiasce

**Comment by simontesar** [comment]
Tests green now

**Comment by simontesar** [comment]
Is there anything else I need to do to get this merged @viglesiasce ?

**Comment by ghost** [comment]
Can we get `.server.affinity` passed through without the hard/soft approach?  We've seen cases where more nuanced config is necessary in this map. Thanks!

**Comment by viglesiasce** [comment]
@josdotso the issue with passing it straight through is that you cannot template it then.

@simt2 what do you think about passing it straight through from values to the manifests?

**Comment by simontesar** [comment]
@josdotso @viglesiasce I was adhering to existing patterns but I agree that passing through `affinity` allows for more flexibility, similar to `nodeSelector` and `tolerations`. I will pass it through and add examples for the classic hard and soft approaches. 

Edit: But afaik it is not possible to reference the release name in `values.yml`, so how would we achieve anti-affinity with only pods of the actual release and not all of the same app type?

**Review by viglesiasce — APPROVED** [review]

**Comment by viglesiasce** [comment]
@simt2 lets get this first implementation in as is since it satisfies the use case ATM. in a future update lets think about how we could achieve this in a v1.0.0 version of this chart.

/lgtm

**Pull request merged**